src/action-ros-ci.ts

```diff
@@ -268,8 +284,19 @@ async function run() {
 			);
 		}
 
-		// Reset colcon configuration.
-		await io.rmRF(path.join(os.homedir(), ".colcon"));
+		// Reset colcon configuration and create defaults file if one was provided.
+		const colconHome = path.join(os.homedir(), ".colcon");
```

Comment:
I think that I would prefer to write this file into a tmp dir, maybe the same place we create the `.repos` files, then use `COLCON_DEFAULTS_FILE` environment variable to specify the file, rather than touching `~/.colcon` - what do you think?

Reply:
It was simpler to do it like this since there was no need to set any env var.

> rather than touching `~/.colcon`

You want to avoid possibly polluting or affecting the global colcon config for any subsequent command/action? In any case I don't mind; I'll switch!

Reply:
> You want to avoid possibly polluting or affecting the global colcon config for any subsequent command/action?

Yes, that is what I was thinking . It's probably not a big deal, I can't think of a specific case - but I'm leaning slightly towards "better safe than sorry" with it

Reply:
Yeah!

Changed in 6f40c4711299d6d15c4628c46d5e46d01b102879